client/webserver/site/src/js/app.js

```diff
@@ -346,6 +353,27 @@ export default class Application {
     }
     return true
   }
+
+  /**
+   * Close all wallets if they are opening, if there is an error while closing wont logout,
+   * if successful will delete auth cookies and reload the page
+   */
+  async signOut () {
+    try {
+      const closeWallet = async (asset) => {
+        if (asset.wallet && asset.wallet.open) {
+          const res = await postJSON('/api/closewallet', { assetID: asset.id })
+          if (!this.checkResponse(res)) throw res.msg
+        }
+      }
+      await Promise.all(
+        Object.keys(this.assets).map((assetID) => closeWallet(this.assets[assetID]))
+      )
+      State.removeCookie(authCK)
+      window.location.reload()
+    } catch (error) {
+    }
+  }
```

Comment:
This is about how I 'd do this if I was doing it in javascript, but unfortunately, we need to handle this server-side. The web server should be deleting the auth cookie, and calling a new core method, `(*Core).Logout`, that will

1. Check for active orders. If there are active orders, then return an error. Alternatively, we could lock the wallets and accounts that are not involved in trades, but still return an error. Not sure which is better. 
2. Call `(*dexAccount).lock` for each account in the `conns` map.

This will require a new `WebServer` endpoint, `(*WebServer).apiLogout` at `/logout`.